nncf/experimental/onnx/graph/onnx_graph.py

```diff
@@ -0,0 +1,156 @@
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+      http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+
+from typing import Callable
+from typing import Dict
+from typing import List
+
+import onnx
+from onnx import NodeProto  # pylint: disable=no-name-in-module
+from onnx import ValueInfoProto  # pylint: disable=no-name-in-module
+import numpy as np
+
+
+# pylint: disable=no-member
+
+class ONNXGraph:
+    """
+    The class stores onnx model and provides the interface to interact with it.
+    """
+
+    def __init__(self, onnx_model: onnx.ModelProto):
+        self.onnx_model = onnx_model
+        self.model_with_shapes = onnx.shape_inference.infer_shapes(self.onnx_model)
+        self.activations_tensors = self.model_with_shapes.graph.value_info
+        inputs = self.model_with_shapes.graph.input
+        outputs = self.model_with_shapes.graph.output
+        self.activations_tensors.extend(inputs)
+        self.activations_tensors.extend(outputs)
+
+    def get_all_nodes(self) -> List[NodeProto]:
+        """
+        Returns all nodes of onnx model.
+        """
+        return self.onnx_model.graph.node
+
+    def get_model_inputs(self) -> List[ValueInfoProto]:
+        """
+        Returns model inputs.
+        """
+        return list(self.onnx_model.graph.input)
+
+    def get_model_outputs(self) -> List[ValueInfoProto]:
+        """
+        Returns model outputs.
+        """
+        return list(self.onnx_model.graph.output)
+
+    def get_nodes_by_output(self, output_name: str) -> List[NodeProto]:
+        """
+        Returns all nodes that have output with the name 'output_name'.
+        """
+        return self._get_nodes_by_lambda(output_name, lambda node: node.output)
+
+    def get_nodes_by_input(self, input_name: str) -> List[NodeProto]:
+        """
+        Returns all nodes that have input with the name 'input_name'.
+        """
+        return self._get_nodes_by_lambda(input_name, lambda node: node.input)
+
+    def _get_nodes_by_lambda(self, name: str, func: Callable[[NodeProto], List[NodeProto]]):
+        output = []
+        graph = self.onnx_model.graph
+        for node in graph.node:
+            if name in func(node) or name == func(node):
+                output.append(node)
+        return output
+
+    def get_node_edges(self, node_name: str) -> Dict[str, List[ValueInfoProto]]:
+        """
+        Returns node input and output edges.
+        """
+        graph = self.onnx_model.graph
+        for node in graph.node:
+            if node.name == node_name:
+                return {'input': list(node.input),
```

Comment:
would be more cool with dataclass instead of this dict, but AFAIK we still have low python version to use it

Reply:
Unfortunately, we support python3.6+, while dataclasses are supported by python3.7+. We don't have a choice here.